synapse/cortex.py

```diff
@@ -2549,7 +2550,12 @@ async def _addStormPkg(self, pkgdef):
         name = pkgdef.get('name')
         olddef = self.pkghive.get(name, None)
         if olddef is not None:
-            await self._dropStormPkg(olddef)
+            if s_hashitem.hashitem(pkgdef) != s_hashitem.hashitem(olddef):
+                await self._dropStormPkg(olddef)
+            else:
+                # if it's the same we can just bounce
+                await self.fire('core:pkg:onload:complete', pkg=name)
```

Comment:
Should we actually be firing this even though we didn't run the onload? Also `loadStormPkg` only fires this if there is actually an `onload` defined so we should at least check that

Reply:
Good point. It made sense initially but probably better to not. I'll just remove this.